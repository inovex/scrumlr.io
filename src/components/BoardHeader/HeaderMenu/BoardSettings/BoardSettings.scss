@import "constants/style";

$header-menu__item-height: 48px;
$header-menu__item-border-width: 1px;
$header-menu-border-bottom-color: #efefef;
$header-menu-toggle-background-color: #c8c8c8;
$header-menu-icon-width: 79px;
$header-menu-icon-height: 24px;
$header-menu-toggle-button-height: 12px;
$header-menu-toggle-button-width: 12px;
$header-menu-toggle-height: 3px;
$header-menu-button-height: 24px;
$header-menu-button-width: 24px;

.board-settings {
  position: relative;
  height: 64px;
  background-color: var(--accent-color);
}

.board-settings > form {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-left: $padding--default;
  padding-right: $padding--default;
}

.board-settings__board-name {
  height: $header-menu-button-height;
  flex-grow: 1;
  color: $color-white;
  background-color: transparent;
  font-size: $text-size--medium;
  font-weight: bold;
  letter-spacing: $letter-spacing--small;
  line-height: $line-height--medium;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  border-top: none;
  border-right: none;
  border-bottom: $header-menu__item-border-width dashed $color-white;
  border-left: none;
  border-radius: 0;
  -webkit-border-radius: 0;
  outline: 0;
  padding: 0;
  transition: border-color 0.1s linear;
  caret-color: $color-white;

  &::placeholder {
    color: $color-white;
    opacity: 0.85;
  }

  &:disabled {
    border-color: transparent;
    opacity: 1;
  }
}

.board-settings__edit-button {
  margin-left: $margin--default;
  height: $header-menu-button-height;
  box-shadow: 0 0 0 0.1em $color-white;
  border-color: transparent;
  border-radius: 32px;
  background-color: transparent;
  color: $color-white;
  cursor: pointer;
  outline: none;
  transition: all 0.1s ease-in-out;

  &:focus-visible {
    background-color: $color-white;
    color: var(--accent-color);
  }

  &:hover {
    background-color: $color-white;
    color: var(--accent-color);
    transform: scale(1.06);
  }

  &:active {
    transform: scale(1);
  }
}

[theme="dark"] {
  .header-menu {
    box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
  }
}
