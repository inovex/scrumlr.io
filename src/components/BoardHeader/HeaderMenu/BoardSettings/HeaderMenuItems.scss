@import "constants/style";

$header-menu-color--light: $color-backlog-blue;
$header-menu-color--dark: $color-planning-pink;
$header-menu__item-height: 48px;
$header-menu__item-border-width: 1px;
$header-menu-hover-color--dark: #3d4555;
$header-menu-hover-color--light: #f3f3f3;
$header-menu-border-bottom-color: #efefef;
$header-menu-toggle-background-color: #c8c8c8;
$header-menu-icon-width: 79px;
$header-menu-icon-height: 24px;
$header-menu-toggle-button-height: 12px;
$header-menu-toggle-button-width: 12px;
$header-menu-toggle-height: 3px;
$header-menu-button-height: 24px;
$header-menu-button-width: 24px;

.header-menu__export-container {
  height: 0;
  width: 100%;
  background: $color-white;
  transition: height 0.5s ease-in-out;
  position: relative;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  row-gap: 10px;
}
.header-menu__export-container--visible {
  box-sizing: border-box;
  height: 90px;
  border-bottom: $header-menu__item-border-width solid $header-menu-border-bottom-color;
}

.header-menu__info {
  position: relative;
  height: 86px;
  background-color: $header-menu-color--light;
}

.info__board-name {
  position: absolute;
  top: 20px;
  left: 20px;
  height: 19px;
  width: 180px;
  color: $color-white;
  font-size: 18px;
  font-weight: bold;
  letter-spacing: 0.25px;
  line-height: 18px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  background-color: transparent;
  border: none;
  border-bottom: $header-menu__item-border-width dashed $color-white;
  padding: 0 0 5px 0;
  outline: 0;
  transition: border-color 0.1s linear;
}
.info__board-name:disabled {
  border-color: transparent;
}

.info__access-mode {
  position: absolute;
  top: 44px;
  left: 20px;
  height: 18px;
  width: max-content;
  border: none;
  background-color: transparent;
  text-align: left;
  color: $color-white;
  opacity: 0.8;
  font-size: 13px;
  letter-spacing: 0.27px;
  line-height: 18px;
  cursor: pointer;
  border-bottom: $header-menu__item-border-width dashed $color-white;
  padding: 0 0 5px 0;
  transition: border-color 0.1s linear;
}
.info__access-mode:disabled {
  cursor: default;
  border-color: transparent;
}
.info__access-mode-lock {
  display: inline-block;
  position: relative;
  width: 10px;
  height: 10px;
  border: 2px solid $color-white;
  background-color: $color-white;
  opacity: 0.8;
  border-radius: 3px;
  margin-right: 5px;
}
.info__access-mode-lock::before {
  content: "";
  position: absolute;
  height: 6px;
  width: 6px;
  background-color: $header-menu-color--light;
  border-radius: 3px;
}

.info__access-mode-lock::after {
  position: absolute;
  content: "";
  width: 6px;
  height: 6px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  border: $header-menu__item-border-width solid $color-white;
  border-bottom: 0;
  top: -7px;
  left: 0;
  transition: transform 0.3s linear;
}
.info__access-mode-lock--unlocked::after {
  transform: rotate(-45deg) translate(0, -1px);
}

.info__edit-button {
  position: absolute;
  height: $header-menu-button-height;
  width: 50px;
  box-shadow: 0 0 0 0.1em #f9fafb;
  border-color: transparent;
  border-radius: 32px;
  background-color: transparent;
  color: $color-white;
  cursor: pointer;
  top: 18px;
  left: 212px;
  transition: scale 0.1s ease-in-out;
}
.info__edit-button:focus {
  outline: 0;
}
.info__edit-button:active {
  scale: 0.95;
}

.header-menu__item {
  height: $header-menu__item-height;
  background-color: $color-white;
  transition: background-color 0.15s ease-in-out;
  box-sizing: border-box;
  border-bottom: $header-menu__item-border-width solid $header-menu-border-bottom-color;
}
.header-menu__item:hover {
  background-color: $header-menu-hover-color--light;
}

.item-button__icon {
  width: $header-menu-icon-width;
  height: $header-menu-icon-height;
  color: #a3a6aa;
  vertical-align: middle;
}

.item-button__label {
  height: $header-menu-button-height;
  width: 180px;
  color: $color-black;
  font-size: 14px;
  letter-spacing: 0.5px;
  line-height: $header-menu-button-height;
  vertical-align: middle;
  cursor: pointer;
}

.item-button__toggle-switch-container {
  width: $header-menu-icon-width;
  height: $header-menu__item-height;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  vertical-align: middle;
}
.item-button__toggle-switch {
  width: $header-menu-button-width;
  height: $header-menu-button-height;
  position: relative;
  display: inline-block;
}
.item-button__toggle-switch::after {
  content: "";
  height: $header-menu-toggle-button-height;
  width: $header-menu-toggle-button-width;
  position: absolute;
  border-radius: 100%;
  top: calc(50% - #{$header-menu-toggle-button-height} / 2);
  transition: all 0.25s linear;
}
.item-button__toggle-switch::before {
  content: "";
  height: $header-menu-toggle-height;
  width: 24px;
  background-color: $header-menu-toggle-background-color;
  position: absolute;
  border-radius: $header-menu-toggle-height;
  top: calc(50% - #{$header-menu-toggle-height} / 2);
}
.item-button__toggle-switch--left::after {
  left: 0;
  background-color: $color-backlog-blue;
}
.item-button__toggle-switch--right::after {
  left: $header-menu-toggle-button-width;
  background-color: $color-planning-pink;
}

.header-menu__qrcode-container {
  height: 0;
  width: 100%;
  background-color: $color-white;
  transition: height 0.5s ease-in-out;
  position: relative;
  overflow: hidden;
}
.header-menu__qrcode-container--visible {
  box-sizing: border-box;
  height: 324px;
  border-bottom: $header-menu__item-border-width solid $header-menu-border-bottom-color;
}
.qrcode-container__qrcode {
  position: absolute;
  top: 13px;
  left: 13px;
}

.qrcode-container__copy-to-clipboard {
  position: absolute;
  height: $header-menu-button-height;
  width: 144px;
  border: $header-menu__item-border-width solid $color-black;
  border-radius: 32px;
  background-color: transparent;
  color: $color-black;
  cursor: pointer;
  left: 72px;
  top: 288px;
}
.qrcode-container__copy-to-clipboard:focus {
  outline: 0;
}
.qrcode-container__copy-to-clipboard:active {
  scale: 0.95;
}

.header-menu__delete-container {
  height: 0;
  width: 100%;
  transition: height 0.5s ease-in-out;
  background-color: $color-white;
  position: relative;
  overflow: hidden;
}

.header-menu__delete-container--visible {
  box-sizing: border-box;
  height: 120px;
}
.delete-container__warning-label {
  position: absolute;
  top: 13px;
  left: 13px;
  color: $color-waring-red;
}
.delete-container__delete-board {
  position: absolute;
  top: 80px;
  left: 72px;
  height: $header-menu-button-height;
  width: 144px;
  border: $header-menu__item-border-width solid $color-waring-red;
  border-radius: 32px;
  background-color: transparent;
  color: $color-waring-red;
  cursor: pointer;
}
.delete-container__delete-board :focus {
  outline: 0;
}
.delete-container__delete-board :active {
  scale: 0.95;
}

@media (prefers-color-scheme: dark) {
  .header-menu {
    box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
  }
  .header-menu__info {
    background-color: $header-menu-color--dark;
  }
  .header-menu__delete-container {
    background-color: $color-dark-mode-note;
  }
  .item-button__label {
    color: $color-white;
  }
  .header-menu__item {
    background-color: $color-dark-mode-note;
  }
  .header-menu__item:hover {
    background-color: #3d4555;
  }
  .header-menu__export-container {
    background-color: $color-dark-mode-note;
  }
}
