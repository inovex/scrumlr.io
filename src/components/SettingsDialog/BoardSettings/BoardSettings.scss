@import "src/constants/style";
@import "../SettingsDialog";

.board-settings__container-wrapper {
  overflow-y: auto;
  padding-right: $padding--small;
  margin-right: calc(-1 * ($padding--small + 10px));
  padding-bottom: $settings-dialog-container--bottom;

  &:focus-visible {
    outline: 2px dashed rgba(var(--accent-color-rgb), 0.5);
    border-radius: 8px;
  }

  @include scrollbar();
}

@media screen and (min-width: 920px) {
  .board-settings__container-wrapper {
    padding-right: $padding--medium;
    margin-right: calc(-1 * ($padding--medium + 10px));
  }
}

.board-settings__container {
  display: flex;
  flex-direction: column;
  gap: 32px;
  width: 100%;
  max-width: calc(100vw - (2 * #{$settings-dialog-container--sides-small}));
  padding-bottom: $padding--small;
}

@media screen and (min-width: 450px) {
  .board-settings__container {
    max-width: calc(380px - (2 * #{$settings-dialog-container--sides-small}));
  }
}

@media screen and (min-width: 920px) {
  .board-settings__container {
    max-width: calc(544px - (2 * #{$settings-dialog-container--sides-large}));
  }
}

.board-settings__board-name-button {
  justify-content: space-between;
  cursor: text;
}

.board-settings__board-name-button span {
  cursor: inherit;
}

.board-settings__board-name-button_input {
  background-color: transparent;
  border: none;
  padding: $padding--default $padding--medium $padding--default $padding--default;
  font-weight: bold;
  letter-spacing: $letter-spacing--medium;
  font-size: $text-size--medium;
  text-align: right;
  max-width: 150px;
  cursor: inherit;
  text-overflow: ellipsis;
}

.board-settings__board-name-button_input:focus-visible {
  outline: none;
}

.board-settings__board-name-button_input::placeholder {
  font-weight: bold;
  letter-spacing: $letter-spacing--medium;
  font-size: $text-size--medium;
  color: $color-placeholder;
}

[theme="dark"] {
  .board-settings__board-name-button_input {
    color: $color-white;
  }
  .board-settings__board-name-button_input::placeholder {
    color: $color-placeholder-dark;
  }
}

.board-settings__group-and-button {
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.board-settings__policy-button {
  display: flex;
  justify-content: space-between;
}

.board-settings__policy-button span {
  cursor: inherit;
  flex: 1;
}

.board-settings__policy-button_value {
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: center;
}

.board-settings__policy-button_value span {
  padding: $padding--default $padding--medium $padding--default 4px;
  font-weight: bold;
  letter-spacing: $letter-spacing--medium;
  font-size: $text-size--medium;
  letter-spacing: 0.5px;
  color: $color-placeholder;
}

@media screen and (max-width: 920px) {
  .board-settings__policy-button_value span {
    max-width: min-content;
    text-align: right;
  }
}

[theme="dark"] {
  .board-settings__policy-button_value span {
    color: $color-placeholder-dark;
  }
}

.board-settings__policy-button_value svg {
  width: $icon--medium;
  height: $icon--medium;
  color: var(--accent-color);
}

.board-settings__password-button {
  justify-content: space-between;
  padding-right: 24px;
  position: relative;
  cursor: text;
}

.board-settings__password-button span {
  cursor: inherit;
}

.board-settings__password-button_value {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.board-settings__password-button_value svg {
  width: $icon--medium;
  height: $icon--medium;
}

.board-settings__show-password-button--enabled {
  color: var(--accent-color);
  cursor: pointer;
  transition: all 0.08s ease-out;

  &:hover {
    filter: brightness(1.4);
    transform: scale(1.1);
  }

  &:active {
    transform: scale(1);
  }

  &:focus-visible {
    transform: scale(1.1);
  }
}

.board-settings__show-password-button--disabled {
  color: $color-placeholder;
  cursor: default;
}

[theme="dark"] {
  .board-settings__show-password-button--disabled {
    color: $color-placeholder-dark;
  }
}

.board-settings__password-button_value-input {
  background-color: transparent;
  border: none;
  padding: $padding--default;
  font-weight: bold;
  letter-spacing: $letter-spacing--medium;
  font-size: $text-size--medium;
  text-align: right;
  max-width: 120px;
}

[theme="dark"] {
  .board-settings__password-button_value-input {
    color: $color-white;
  }
}

.board-settings__password-button_value-input:focus-visible {
  outline: none;
}

.board-settings__password-management-button,
.board-settings__password-input-hint {
  display: flex;
  align-items: center;
  padding: 0 $padding--medium;
  height: 30px;
  border: none;
  background-color: inherit;
  color: var(--accent-color);
  align-self: center;
}

.board-settings__password-management-button {
  cursor: pointer;
  transition: all 0.08s ease-out;
  outline: none;

  &:hover {
    filter: brightness(1.4);
    transform: scale(1.01);
  }

  &:active {
    transform: scale(1);
  }

  &:focus-visible {
    text-decoration: underline;
  }
}

.board-settings__remove-protection-button svg {
  width: $icon--medium;
  height: $icon--medium;
  color: var(--accent-color);
}

.board-settings__generate-password-button svg {
  width: $icon--small;
  height: $icon--small;
  margin: 0 4px 2px 3px;
  color: var(--accent-color);
}

.board-settings__password-management-text {
  padding-left: 4px;
  font-weight: bold;
  font-size: $text-size--medium;
  letter-spacing: $letter-spacing--medium;
}

.board-settings__delete-button span {
  color: $color-retro-red;
  font-weight: bold;
}

[theme="dark"] {
  .board-settings__delete-button span {
    color: $color-retro-red;
  }
}

.board-settings__delete-button svg {
  color: $color-retro-red;
}

.board-settings__confirmation-dialog {
  z-index: 0;
}

.button--centered {
  align-self: center;
}
