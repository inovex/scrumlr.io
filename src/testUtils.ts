export type AssertTypeEqual<T, Expected> =
    T extends Expected
        ? (Expected extends T ? true : never)
        : never;