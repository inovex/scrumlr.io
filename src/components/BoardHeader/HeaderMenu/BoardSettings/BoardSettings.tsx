import "./BoardSettings.scss";
import {Dispatch, SetStateAction, useEffect, useState} from "react";
import store, {useAppSelector} from "store";
import {ApplicationState} from "types";
import {Actions} from "store/action";
import {useTranslation} from "react-i18next";
import {DEFAULT_BOARD_NAME} from "../../../../constants/misc";

export type BoardSettingsProps = {
  activeEditMode: boolean;
  currentUserIsModerator: boolean;
  setActiveEditMode: Dispatch<SetStateAction<boolean>>;
};

export const BoardSettings = (props: BoardSettingsProps) => {
  const {t} = useTranslation();
  const state = useAppSelector((applicationState: ApplicationState) => ({
    board: applicationState.board.data!,
  }));

  const [boardName, setBoardName] = useState(state.board.name ?? "");

  useEffect(() => {
    setBoardName(state.board.name ?? "");
  }, [state.board.name]);

  const onSubmit = () => {
    if (props.activeEditMode && state.board!.name !== boardName) {
      store.dispatch(Actions.editBoard({name: boardName}));
    }
    props.setActiveEditMode(!props.activeEditMode);
  };

  return (
    <li className="board-settings">
      <form
        onSubmit={(e) => {
          e.preventDefault();
          onSubmit();
        }}
        onKeyDown={(e) => {
          if (e.key === "Enter") onSubmit();
        }}
      >
        <input
          className="board-settings__board-name"
          value={boardName}
          placeholder={DEFAULT_BOARD_NAME}
          disabled={!props.activeEditMode}
          onChange={(e) => setBoardName(e.target.value)}
          ref={(input) => {
            if (!input?.disabled) input?.focus();
          }}
          onFocus={(e) => {
            e.target.placeholder = "";
            if (boardName !== "") {
              e.target.select();
            }
          }}
          onBlur={(e) => {
            e.target.placeholder = DEFAULT_BOARD_NAME;
          }}
        />

        {props.currentUserIsModerator && (
          <button type="submit" className="board-settings__edit-button">
            {props.activeEditMode ? t("BoardSettings.save") : t("BoardSettings.edit")}
          </button>
        )}
      </form>
    </li>
  );
};
