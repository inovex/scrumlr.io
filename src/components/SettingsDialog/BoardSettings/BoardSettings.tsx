import classNames from "classnames";
import {useTranslation} from "react-i18next";
import {ChangeEvent, useEffect, useState} from "react";
import {Actions} from "store/action";
import store, {useAppSelector} from "store";
import {ReactComponent as SetPolicyIcon} from "assets/icon-lock.svg";
import {ReactComponent as DeleteIcon} from "assets/icon-delete.svg";
import {ReactComponent as VisibleIcon} from "assets/icon-visible.svg";
import {ReactComponent as HiddenIcon} from "assets/icon-hidden.svg";
import {ReactComponent as RefreshIcon} from "assets/icon-refresh.svg";
import {DEFAULT_BOARD_NAME, MIN_PASSWORD_LENGTH, PLACEHOLDER_PASSWORD, TOAST_TIMER_SHORT} from "constants/misc";
import {Toast} from "utils/Toast";
import {generateRandomString} from "utils/random";
import {Toggle} from "components/Toggle";
import {ConfirmationDialog} from "components/ConfirmationDialog";
import {isEqual} from "underscore";
import {SettingsButton} from "../Components/SettingsButton";
import {SettingsInput} from "../Components/SettingsInput";
import "./BoardSettings.scss";

export const BoardSettings = () => {
  const {t} = useTranslation();

  const state = useAppSelector(
    (applicationState) => ({
      board: applicationState.board.data!,
      me: applicationState.participants?.self,
      currentUserIsModerator: applicationState.participants?.self.role === "OWNER" || applicationState.participants?.self.role === "MODERATOR",
    }),
    isEqual
  );

  const [boardName, setBoardName] = useState<string>(state.board.name ?? "");
  const [password, setPassword] = useState<string>("");
  const [showPassword, setShowPassword] = useState<boolean>(false);
  const [showConfirmationDialog, setShowConfirmationDialog] = useState<boolean>(false);
  const [isProtectedOnInitialSettingsOpen, setIsProtectedOnInitialSettingsOpen] = useState(state.board.accessPolicy === "BY_PASSPHRASE");
  const [isProtected, setIsProtected] = useState(state.board.accessPolicy === "BY_PASSPHRASE");

  const isByInvite = state.board.accessPolicy === "BY_INVITE";

  useEffect(() => {
    setBoardName(state.board.name ?? "");
  }, [state.board.name]);

  useEffect(() => {
    setIsProtected(state.board.accessPolicy === "BY_PASSPHRASE");
  }, [state.board.accessPolicy]);

  const handleSetPassword = (newPassword: string) => {
    setPassword(newPassword);
    if (newPassword.length >= MIN_PASSWORD_LENGTH) {
      store.dispatch(Actions.editBoard({accessPolicy: "BY_PASSPHRASE", passphrase: newPassword}));
      navigator.clipboard.writeText(newPassword).then(() => Toast.success({title: t("Toast.passwordCopied"), autoClose: TOAST_TIMER_SHORT}));
      setIsProtected(true);
    } else if (isProtected || isProtectedOnInitialSettingsOpen) {
      store.dispatch(Actions.editBoard({accessPolicy: "PUBLIC"}));
      setIsProtectedOnInitialSettingsOpen(false);
      setIsProtected(false);
      Toast.info({title: t("Toast.boardMadePublic"), autoClose: TOAST_TIMER_SHORT});
    }
  };

  const getPasswordManagementButton = () => {
    if (isProtected) {
      return (
        <button
          className="board-settings__password-management-button board-settings__remove-protection-button button--centered"
          onClick={() => {
            handleSetPassword("");
          }}
        >
          <SetPolicyIcon />
          <span className="board-settings__password-management-text">{t("BoardSettings.SetAccessPolicyOpen")}</span>
        </button>
      );
    }
    if (!password) {
      return (
        <button
          className="board-settings__password-management-button board-settings__generate-password-button"
          onClick={() => {
            const pw = generateRandomString();
            handleSetPassword(pw);
          }}
        >
          <RefreshIcon />
          <span className="board-settings__password-management-text">{t("BoardSettings.generatePassword")}</span>
        </button>
      );
    }
    return <span className="board-settings__password-input-hint board-settings__password-management-text">{t("BoardSettings.SecurePasswordHint")}</span>;
  };

  const getPasswordVisibilityButton = () =>
    showPassword ? (
      <VisibleIcon className="board-settings__show-password-button--enabled" onClick={() => setShowPassword(false)} />
    ) : (
      <HiddenIcon
        className={password ? "board-settings__show-password-button--enabled" : "board-settings__show-password-button--disabled"}
        onClick={() => password && setShowPassword(true)}
      />
    );

  const getAccessPolicyTitle = () => {
    if (isByInvite)
      return (
        <>
          <span>{t("AccessPolicySelection.manualVerificationTitle")}</span>
          <SetPolicyIcon />
        </>
      );
    return !isProtected ? (
      <span>{t("AccessPolicySelection.publicTitle")}</span>
    ) : (
      <>
        <span>{t("AccessPolicySelection.byPassphraseTitle")}</span>
        <SetPolicyIcon />
      </>
    );
  };

  return (
    <div className={classNames("settings-dialog__container", "accent-color__backlog-blue")}>
      <header className="settings-dialog__header">
        <h2 className="settings-dialog__header-text"> {t("SettingsDialog.BoardSettings")}</h2>
      </header>
      <div className="board-settings__container-wrapper">
        <div className="board-settings__container">
          <SettingsInput
            value={boardName}
            id="boardSettingsBoardName"
            label={t("BoardSettings.BoardName")}
            onChange={(e: ChangeEvent<HTMLInputElement>) => setBoardName(e.target.value)}
            submit={() => store.dispatch(Actions.editBoard({name: boardName}))}
            disabled={!state.currentUserIsModerator}
            placeholder={DEFAULT_BOARD_NAME}
            maxLength={128}
          />

          <div className="board-settings__group-and-button">
            <div className="settings-dialog__group">
              <SettingsButton className="board-settings__policy-button" label={t("BoardSettings.AccessPolicy")} disabled>
                <div className="board-settings__policy-button_value">{getAccessPolicyTitle()}</div>
              </SettingsButton>
              {!isByInvite && state.currentUserIsModerator && (
                <>
                  <hr className="settings-dialog__separator" />
                  <SettingsInput
                    id="boardSettingsPassword"
                    label={t("BoardSettings.Password")}
                    value={password}
                    onChange={(e) => {
                      setPassword(e.target.value);
                    }}
                    submit={() => handleSetPassword(password)}
                    type={showPassword ? "text" : "password"}
                    placeholder={!password && !isProtected ? undefined : PLACEHOLDER_PASSWORD}
                  >
                    {password && getPasswordVisibilityButton()}
                  </SettingsInput>
                </>
              )}
            </div>

            {!isByInvite && state.currentUserIsModerator && getPasswordManagementButton()}
          </div>
          {state.currentUserIsModerator && (
            <>
              <div className="settings-dialog__group">
                <SettingsButton
                  data-testid="author"
                  className="board-settings__show-author-button"
                  label={t("BoardSettings.ShowAuthorOption")}
                  onClick={() => {
                    store.dispatch(Actions.editBoard({showAuthors: !state.board.showAuthors}));
                  }}
                  role="switch"
                  aria-checked={state.board.showAuthors}
                >
                  <div className="board-settings__show-author-value">
                    <Toggle active={state.board.showAuthors} />
                  </div>
                </SettingsButton>
                <hr className="settings-dialog__separator" />
                <SettingsButton
                  data-testid="notes"
                  className="board-settings__show-notes-button"
                  label={t("BoardSettings.ShowOtherUsersNotesOption")}
                  onClick={() => store.dispatch(Actions.editBoard({showNotesOfOtherUsers: !state.board.showNotesOfOtherUsers}))}
                  role="switch"
                  aria-checked={state.board.showNotesOfOtherUsers}
                >
                  <div className="board-settings__show-notes-value">
                    <Toggle active={state.board.showNotesOfOtherUsers} />
                  </div>
                </SettingsButton>
                <SettingsButton
                  data-testid="reactions"
                  className="board-settings__show-reactions-button"
                  label={t("BoardSettings.ShowNoteReactionsOptions")}
                  onClick={() => store.dispatch(Actions.editBoard({showNoteReactions: !state.board.showNoteReactions}))}
                  role="switch"
                  aria-checked={state.board.showNoteReactions}
                >
                  <div className="board-settings__show-reactions-value">
                    <Toggle active={state.board.showNoteReactions} />
                  </div>
                </SettingsButton>
                <hr className="settings-dialog__separator" />
                <SettingsButton
                  data-testid="columns"
                  className="board-settings__show-columns-button"
                  label={t("BoardSettings.ShowHiddenColumnsOption")}
                  onClick={() => store.dispatch(Actions.setShowHiddenColumns(!state.me?.showHiddenColumns))}
                  role="switch"
                  aria-checked={!!state.me?.showHiddenColumns}
                >
                  <div className="board-settings__show-columns-value">
                    <Toggle active={!!state.me?.showHiddenColumns} />
                  </div>
                </SettingsButton>
                <hr className="settings-dialog__separator" />
                <SettingsButton
                  data-testid="note-repositioning"
                  className="board-settings__allow-note-repositioning-button"
                  label={t("BoardSettings.AllowNoteRepositioningOption")}
                  onClick={() => store.dispatch(Actions.editBoard({allowStacking: !state.board.allowStacking}))}
                  role="switch"
                  aria-checked={state.board.allowStacking}
                >
                  <div className="board-settings__allow-note-repositioning-value">
                    <Toggle active={state.board.allowStacking} />
                  </div>
                </SettingsButton>
              </div>

              <SettingsButton
                className={classNames("board-settings__delete-button")}
                label={t("BoardSettings.DeleteBoard")}
                onClick={() => setShowConfirmationDialog(true)}
                icon={DeleteIcon}
              />
            </>
          )}

          {showConfirmationDialog && (
            <ConfirmationDialog
              title={t("ConfirmationDialog.deleteBoard")}
              onAccept={() => store.dispatch(Actions.deleteBoard())}
              onDecline={() => setShowConfirmationDialog(false)}
              icon={DeleteIcon}
              warning
            />
          )}
        </div>
      </div>
    </div>
  );
};
