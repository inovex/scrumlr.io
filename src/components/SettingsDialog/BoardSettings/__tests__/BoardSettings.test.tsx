import {Provider} from "react-redux";
import {render} from "testUtils";
import getTestStore from "utils/test/getTestStore";
import {BoardSettings} from "../BoardSettings";

const createBoardSettings = () => (
  <Provider store={getTestStore()}>
    <BoardSettings />
  </Provider>
);

describe("BoardSettings", () => {
  test("should match snapshot", () => {
    const {container} = render(createBoardSettings());
    expect(container.firstChild).toMatchSnapshot();
  });
});
