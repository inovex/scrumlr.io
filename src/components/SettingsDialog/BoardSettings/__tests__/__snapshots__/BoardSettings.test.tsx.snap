// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardSettings should match snapshot 1`] = `
<div
  class="settings-dialog__container accent-color__backlog-blue"
>
  <header
    class="settings-dialog__header"
  >
    <h2
      class="settings-dialog__header-text"
    >
       
      SettingsDialog.BoardSettings
    </h2>
  </header>
  <div
    class="board-settings__container-wrapper"
  >
    <div
      class="board-settings__container"
    >
      <div
        class="settings-input__container"
      >
        <input
          autocomplete="off"
          id="boardSettingsBoardName"
          maxlength="128"
          placeholder="scrumlr.io"
          type="text"
          value="test-board-name"
        />
        <small
          class="settings-input__length"
        >
          15
          /
          128
        </small>
        <label
          for="boardSettingsBoardName"
        >
          BoardSettings.BoardName
        </label>
      </div>
      <div
        class="board-settings__group-and-button"
      >
        <div
          class="settings-dialog__group"
        >
          <button
            aria-label="BoardSettings.AccessPolicy"
            class="settings-option-button settings-option-button--disabled board-settings__policy-button"
            disabled=""
            type="button"
          >
            <span
              class="settings-option-button__label"
            >
              BoardSettings.AccessPolicy
            </span>
            <div
              class="board-settings__policy-button_value"
            >
              <span>
                AccessPolicySelection.publicTitle
              </span>
            </div>
          </button>
          <hr
            class="settings-dialog__separator"
          />
          <div
            class="settings-input__container"
          >
            <input
              autocomplete="off"
              class="settings-input__hidden-placeholder"
              id="boardSettingsPassword"
              placeholder="BoardSettings.Password"
              type="password"
              value=""
            />
            <label
              for="boardSettingsPassword"
            >
              BoardSettings.Password
            </label>
          </div>
        </div>
        <button
          class="board-settings__password-management-button board-settings__generate-password-button"
        >
          <svg>
            icon-refresh.svg
          </svg>
          <span
            class="board-settings__password-management-text"
          >
            BoardSettings.generatePassword
          </span>
        </button>
      </div>
      <div
        class="settings-dialog__group"
      >
        <button
          aria-checked="true"
          aria-label="BoardSettings.ShowAuthorOption"
          class="settings-option-button board-settings__show-author-button"
          data-testid="author"
          role="switch"
          type="button"
        >
          <span
            class="settings-option-button__label"
          >
            BoardSettings.ShowAuthorOption
          </span>
          <div
            class="board-settings__show-author-value"
          >
            <div
              class="toggle toggle--active"
            />
          </div>
        </button>
        <hr
          class="settings-dialog__separator"
        />
        <button
          aria-checked="true"
          aria-label="BoardSettings.ShowOtherUsersNotesOption"
          class="settings-option-button board-settings__show-notes-button"
          data-testid="notes"
          role="switch"
          type="button"
        >
          <span
            class="settings-option-button__label"
          >
            BoardSettings.ShowOtherUsersNotesOption
          </span>
          <div
            class="board-settings__show-notes-value"
          >
            <div
              class="toggle toggle--active"
            />
          </div>
        </button>
        <button
          aria-checked="true"
          aria-label="BoardSettings.ShowNoteReactionsOptions"
          class="settings-option-button board-settings__show-reactions-button"
          data-testid="reactions"
          role="switch"
          type="button"
        >
          <span
            class="settings-option-button__label"
          >
            BoardSettings.ShowNoteReactionsOptions
          </span>
          <div
            class="board-settings__show-reactions-value"
          >
            <div
              class="toggle toggle--active"
            />
          </div>
        </button>
        <hr
          class="settings-dialog__separator"
        />
        <button
          aria-checked="false"
          aria-label="BoardSettings.ShowHiddenColumnsOption"
          class="settings-option-button board-settings__show-columns-button"
          data-testid="columns"
          role="switch"
          type="button"
        >
          <span
            class="settings-option-button__label"
          >
            BoardSettings.ShowHiddenColumnsOption
          </span>
          <div
            class="board-settings__show-columns-value"
          >
            <div
              class="toggle"
            />
          </div>
        </button>
        <hr
          class="settings-dialog__separator"
        />
        <button
          aria-checked="true"
          aria-label="BoardSettings.AllowNoteRepositioningOption"
          class="settings-option-button board-settings__allow-note-repositioning-button"
          data-testid="note-repositioning"
          role="switch"
          type="button"
        >
          <span
            class="settings-option-button__label"
          >
            BoardSettings.AllowNoteRepositioningOption
          </span>
          <div
            class="board-settings__allow-note-repositioning-value"
          >
            <div
              class="toggle toggle--active"
            />
          </div>
        </button>
      </div>
      <button
        aria-label="BoardSettings.DeleteBoard"
        class="settings-option-button board-settings__delete-button"
        type="button"
      >
        <span
          class="settings-option-button__label"
        >
          BoardSettings.DeleteBoard
        </span>
        <svg
          class="settings-option-button__icon"
        >
          icon-delete.svg
        </svg>
      </button>
    </div>
  </div>
</div>
`;
